### PR TITLE
Operations/API clients refactoring

### PR DESCRIPTION
**Pull request by slemrmartin**
- added shared sources api/topo api mixins
- changed topological api client to class (to behave the same like souces api client class)
- removed endpoint_client and processor (not used)
- added `<debug|info|warn|error|fatal>_ext` logger methods to have unified log with operation name

---

Not compatible change, should be released in version 2.0.0

---

[RHCLOUD-9343](https://issues.redhat.com/browse/RHCLOUD-9343)

### CODE REVIEW

File: lib/topological_inventory/providers/common/mixins/sources_api.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
+require "topological_inventory/providers/common/sources_api_client"
+
+module TopologicalInventory
+  module Providers
+    module Common
+      module Mixins
+        module SourcesApi
+          # TODO: Correct?
```
Comment (lindgrenj6) [inline lib/topological_inventory/providers/common/mixins/sources_api.rb]: Curious what this comment is referring to? It's basically only here so we don't return nil when authentications aren't necessary (ie receptor enabled sources).

File: lib/topological_inventory/providers/common/operations/processor.rb [hunk 2]
```diff
@@ -1,135 +0,0 @@
-require "topological_inventory/providers/common/operations/topology_api_client"
-
-module TopologicalInventory
-  module Providers
-    module Common
-      module Operations
-        class Processor
```
Comment (lindgrenj6) [inline lib/topological_inventory/providers/common/operations/processor.rb:7]: So where did this one go? Were we using it before as a base class but now we aren't?

Comment (slemrmartin) [reply lib/topological_inventory/providers/common/operations/processor.rb:7]: The Processor class was created there long time ago but never used as a base class in any repo, so it's redundant

File: lib/topological_inventory/providers/common/mixins/sources_api.rb (hunk 1 above)
Comment (slemrmartin) [reply lib/topological_inventory/providers/common/mixins/sources_api.rb]: emm, unfortunatelly I don't know what I wanted to say by this, I'll remove it :D

Comment (slemrmartin) [reply lib/topological_inventory/providers/common/mixins/sources_api.rb]: I know, it was my comment before internal testing, I forgot to remove it :)

**Review by lindgrenj6 — APPROVED** [review]
LGTM, will merge after rubocop fixes are put in.

**Comment by slemrmartin** [comment]
@lindgrenj6 I'm not planning to "fix" these rubocop issues. The first one in nonsense in the context of code. 
The `CGI.escape` etc is not equivalent to `URI.encode`, I have bad experience in swapping these checks.

**Review by roliveri — APPROVED** [review]

**Pull request merged**